src/opl3.h

```diff
@@ -0,0 +1,135 @@
+//
+// Copyright (C) 2013-2016 Alexey Khokholov (Nuke.YKT)
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+//
+//  Nuked OPL3 emulator.
+//  Thanks:
+//      MAME Development Team(Jarek Burczynski, Tatsuyuki Satoh):
+//          Feedback and Rhythm part calculation information.
+//      forums.submarine.org.uk(carbon14, opl3):
+//          Tremolo and phase generator calculation information.
+//      OPLx decapsulated(Matthew Gambrell, Olli Niemitalo):
+//          OPL2 ROMs.
+//
+// version: 1.7.4
+//
+
+#ifndef OPL_OPL3_H
+#define OPL_OPL3_H
+
```

Comment:
I would suggest renaming the files from `opl3.c` and `opl3.h` to `nukedopl.c` and `nukedopl.h` so it's clear what they are.  The existing AdPlug OPL synth files are somewhat poorly named so let's not continue that tradition!  This would involve updating this `#define` constant too for consistency, however you could leave the `#define` as is if you split out the `class` definition of `CNemuopl` into a separate `.cpp` file as suggested above.  The reason is that if you split out the class definition, then only one `.cpp` file will ever `#include` this file, so the `#define` value is not particularly important in that arrangement.